stan/math/prim/mat/prob/neg_binomial_2_log_glm_lpmf.hpp

```diff
@@ -95,31 +125,32 @@ neg_binomial_2_log_glm_lpmf(const T_y& y, const T_x& x, const T_alpha& alpha,
   if (!include_summand<propto, T_x, T_alpha, T_beta, T_precision>::value)
     return 0.0;
 
-  Array<T_partials_return, Dynamic, 1> y_arr(N, 1);
-  Array<T_partials_return, Dynamic, 1> phi_arr(N, 1);
+  const T_precision_mat& phi_val = value_of(phi);
+  const T_precision_val& phi_arr = phi_val;  // No copy?
```

Comment:
How does this work?

```
  const T_precision_mat& phi_val = value_of(phi);
```

Isn't the return of `value_of` a matrix rather than a reference?   I see that it's `const`---it won't compile without that, but how does the referenced object persist outside of the assignment?